frontends/p4/typeChecking/typeConstraints.cpp

```diff
@@ -111,4 +114,69 @@ void TypeConstraints::dbprint(std::ostream &out) const {
     }
 }
 
+std::string TypeConstraint::localError(Explain *explainer) const {
+    if (errFormat.isNullOrEmpty()) return "";
+
+    std::string message, explanation;
+    boost::format fmt = boost::format(errFormat);
```

Comment:
I am guessing we can not get rid of this particular format call because the input is boost-formatted?

Reply:
Let me check

Reply:
So, yes, the input is just string, so we need to wrap, etc :(

However, I looked around and did some light refactoring removing unnecessary `cstring`'s.

Reply:
Thanks, looks good to me!